### PR TITLE
refactor: change in ITk seed confirmation for high pT muons efficiency 

### PR DESCRIPTION
**Pull request by LuisFelipeCoelho**
As discussed by @noemina in https://indico.cern.ch/event/1258598/contributions/5286608/attachments/2607062/4503967/UpgradeTracking_230308.pdf
and Athena MR https://gitlab.cern.ch:8443/atlas/athena/-/merge_requests/61367 this PR slightly changes the seed confirmation and ITk configuration (z binning and maximum distance between middle and bottom space points) to improve reconstruction for high pT muons

### CODE REVIEW

File: Core/include/Acts/Seeding/SeedFilter.ipp [hunk 1]
```diff
@@ -46,10 +46,13 @@ void SeedFilter<external_spacepoint_t>::filterSeeds_2SpFixed(
             : m_cfg.centralSeedConfirmationRange;
     // set the minimum number of top SP depending on whether the bottom SP is
     // in the central or forward region
-    seedFilterState.nTopSeedConf =
-        bottomSP.radius() > seedConfRange.rMaxSeedConf
-            ? seedConfRange.nTopForLargeR
-            : seedConfRange.nTopForSmallR;
+    if (bottomSP.radius() > seedConfRange.rMaxSeedConf or
+        middleSP.z() < seedFilterState.zMaxSeedConf or
```
Comment (noemina) [inline Core/include/Acts/Seeding/SeedFilter.ipp:50]: Shouldn't this be:
`(zM < seedFilterState.zMaxSeedConf and zM > seedFilterState.zMinSeedConf)` ?

File: Core/include/Acts/Seeding/SeedFinder.ipp [hunk 2]
```diff
@@ -376,7 +380,8 @@ void SeedFinder<external_spacepoint_t, platform_t>::filterCandidates(
     // middle bottom pair if seedConfirmation is false we always ask for at
     // least one compatible top to trigger the filter
     size_t minCompatibleTopSPs = 2;
-    if (!m_config.seedConfirmation or
+    if (!m_config.seedConfirmation or zM < seedFilterState.zMaxSeedConf or
+        zM > seedFilterState.zMinSeedConf or
```
Comment (noemina) [inline Core/include/Acts/Seeding/SeedFinder.ipp:384]: Same comment as above.

File: Core/include/Acts/Seeding/SeedFinderOrthogonal.ipp [hunk 3]
```diff
@@ -339,7 +340,8 @@ void SeedFinderOrthogonal<external_spacepoint_t>::filterCandidates(
     // middle bottom pair if seedConfirmation is false we always ask for at
     // least one compatible top to trigger the filter
     size_t minCompatibleTopSPs = 2;
-    if (!m_config.seedConfirmation or
+    if (!m_config.seedConfirmation or zM < seedFilterState.zMaxSeedConf or
```
Comment (noemina) [inline Core/include/Acts/Seeding/SeedFinderOrthogonal.ipp:343]: Same comment as above.

File: Core/include/Acts/Seeding/SeedFilter.ipp (hunk 1 above)
Comment (LuisFelipeCoelho) [reply Core/include/Acts/Seeding/SeedFilter.ipp]: yes, done 😅

**Comment by LuisFelipeCoelho** [comment]
The change in ITk performance is expected so I replaced the root hash

File: Core/include/Acts/Seeding/SeedFilter.hpp [hunk 4]
```diff
@@ -26,6 +26,12 @@ struct SeedFilterState {
   float zOrigin = 0;
   // number of minimum top SPs in seed confirmation
   size_t nTopSeedConf = 0;
+  // z minimum and maximum of middle component of the seed used to define the
+  // region of the detector for seed confirmation
+  float zMinSeedConf =
+      std::numeric_limits<float>::min();  // Acts::UnitConstants::mm
```
Comment (CarloVarni) [inline Core/include/Acts/Seeding/SeedFilter.hpp:32]: This is a `0`. Is the z-range defined only in the `[0, inf)` range? Or did you mean to use `std::numeric_limits<float>::lowest()`?

Comment (andiwand) [reply Core/include/Acts/Seeding/SeedFilter.hpp:32]: I think this would actually be the smallest positive number greater than 0

Comment (LuisFelipeCoelho) [reply Core/include/Acts/Seeding/SeedFilter.hpp]: that is indeed `lowest`, thank you for noticing

**Comment by LuisFelipeCoelho** [comment]
I guess I can close this PR since it was reverted in Athena @noemina

**Comment by LuisFelipeCoelho** [comment]
Or actually I will keep only the changes from Carlo's comment

**Comment by CarloVarni** [comment]
@LuisFelipeCoelho news on this? 

I see there are changes to the configuration, does that improve timing performance (how much?) while keeping high physics performance?

**Comment by CarloVarni** [comment]
just to confirm then... in Athena these changes produced worse performance? But in ACTS it is ok?

**Comment by LuisFelipeCoelho** [comment]
no, sorry. The seed confirmation changes in the first PR resulted in worse performanc. These changes were reverted in Athena, but they kept the configuration changes, which does not affect performance too much.
Here in acts I only apply the changes that were kept in Athena.

**Pull request merged**